src/dbnode/integration/write_quorum_test.go

```diff
@@ -223,14 +225,14 @@ func makeTestWrite(
 	numShards int,
 	instances []services.ServiceInstance,
 ) (testSetups, closeFn, testWriteFn) {
-
 	nsOpts := namespace.NewOptions()
 	md, err := namespace.NewMetadata(testNamespaces[0],
 		nsOpts.SetRetentionOptions(nsOpts.RetentionOptions().SetRetentionPeriod(6*time.Hour)))
 	require.NoError(t, err)
 
 	nspaces := []namespace.Metadata{md}
-	nodes, topoInit, closeFn := newNodes(t, numShards, instances, nspaces, false)
+	nodes, topoInit, closeFn := newNodes(t, numShards, instances, nspaces, false,
+		bootstrapper.NoOpAllBootstrapperName)
```

Comment:
Any reason why in some cases we can get away with `UninitializedTopology` as final bootstrapper, and in some cases we are still keeping `NoOpAllBootstrapperName`?

Reply:
Uninitialized bootstrapper only comes into play when number of initializing shards is higher than leaving ones:
https://github.com/m3db/m3/blob/a9898e234eba88f3b1b03270d6d53c35a1259f1a/src/dbnode/storage/bootstrap/bootstrapper/uninitialized/source.go#L126-L132

However, there are some tests which doesn't satisfy that and fall into retry loop:
https://github.com/m3db/m3/blob/a9898e234eba88f3b1b03270d6d53c35a1259f1a/src/dbnode/integration/fetch_tagged_quorum_test.go#L138-L143